### ISSUE TITLE
The turret dropper fails to safely lower a turret on a glass panel

### ISSUE THREAD

**Issue by vtipoman** [BEEmod/BEE2-items#1997]
This might be the case with other "pop-up" panels as well
Just to be clear, it was set to 90°.

**Comment by Luke18033** [comment]
Is this in Portal 1 style? Angled panels are quite glitchy there.

**Comment by TeamSpen210** [comment]
Currently it's not smart enough to know about panels and glass.

**Comment by WhoozYerBuddy** [comment]
I think that it should be able to recognize glass/grating, 1/4 walls and partial flat blocks, or as an alternative -   would it be possible to set the height of the turret like you can place the height of the cube under a cube dropper?  That way if something is there to support the turret it would be placed on it, if not it would fall.  If something like a light bridge or angled panel opened above the set height to block the path, the turret would get toppled like it currently does.

**Comment by TeamSpen210** [comment]
Indeed, but I need to write the infrastructure for it to be able to detect surfaces. I can't make it work like the cube dropper item, that doesn't actually do anything anyway.

**Comment by WhoozYerBuddy** [comment]
BTW, the turret dropper is the best new item with this last release.  Fantastic work.  It really expands the potential uses of turrets.  

I understand that changing the height of a cube under a dropper doesn't affect its function.  I was suggesting that as a user-interface rather than an in-game function.  Maybe the little ball/arrow widget on the piston platform would be a better analogy.  It would set the length of the cable in the turret dropper.  The cable would extend that far and release the turret.  If there was a solid object under the turret at that height, the turret would not topple.  If there wasn't anything solid at that level then the turret would fall.  If something solid was in the way of the cable extending to its set distance (light bridge, angled panel, closed solid field, etc) then the turret would topple like it currently does.  I guess it doesn't really matter, though.  If it's easier to code having it detect surfaces that would work well also.

**Comment by TeamSpen210** [comment]
We can't alter the editor like that, mostly just reproduce existing functionality...

**Comment by Luke18033** [comment]
One option for this, which would avoid new compiler logic entirely - just use a VScript to detect if the claw is blocked by something, and release the turret. Maybe have it retract slightly first to prevent it from getting stuck in the floor.

**Comment by Luke18033** [comment]
Basically like the logic piston platforms use to enable the fizzle trigger.
